### PR TITLE
Pickers Demo

### PR DESCRIPTION
**Pull request by yschimke**
Draft of a Pickers Demo

TODO

- [x] Agree menu layout (merge with user interface components)
- [x] Agree functionality - show toast, or show LocalDateTime value on screen
- [x] Agree patterns to demonstrate passing and returning values (dialog?)
- [x] fix navigation bug 12 -> 24 and vice versa

Later

- [ ] Fix the time text overlapping am/pm in a follow up review

### CODE REVIEW

**Comment by yschimke** [comment]
Picker Screen 
image

**Comment by yschimke** [comment]
24 hour
image

**Comment by yschimke** [comment]
12 hour 
image

**Comment by yschimke** [comment]
github images seem to be having issues, I'll edit once they are working again.

**Comment by shumelchyk** [comment]
Any reason why Pickers can't be under User Input Components?
Do you also mind adding screenshots for https://github.com/android/wear-os-samples/issues/216?

**Comment by yschimke** [comment]
@shumelchyk Sure, I just need to change from a Column to SLC for those screens.

**Comment by yschimke** [comment]
Also user interface components screen has a value, that the children change. So that could be confusing.

**Comment by shumelchyk** [comment]
Cool, thanks @yschimke 
Just nit: TimeText overlaps with Picker's AM/PM chip and top label.

> Also user interface components screen has a value, that the children change. So that could be confusing.

Feel free to move this out from parent screen, I agree that it's odd when we add more input components.

**Comment by yschimke** [comment]
@shumelchyk I'll fix TimeText in a follow up, we will need to remove on this screen.

From description:
> Fix the time text overlapping am/pm in a follow up review

File: ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/ui/landing/LandingScreen.kt [hunk 1]
```diff
@@ -67,68 +67,74 @@ fun LandingScreen(
...
 
         // Places both Chips (button and toggle) in the middle of the screen.
-        Column(
-            modifier = Modifier
-                .fillMaxSize()
-                .padding(horizontal = 10.dp),
-            verticalArrangement = Arrangement.spacedBy(4.dp, Alignment.CenterVertically)
+        ScalingLazyColumn(
+            modifier = Modifier.fillMaxSize(),
+            state = ScalingLazyListState(initialCenterItemIndex = 1)
```
Comment (JohnNichol) [inline ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/ui/landing/LandingScreen.kt:72]: These are defaults - we shouldn't need them and so I recommend we remove

File: ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/ui/landing/LandingScreen.kt [hunk 2]
```diff
@@ -67,68 +67,74 @@ fun LandingScreen(
...
-                    Text(
-                        stringResource(R.string.list_of_watches_button_label),
-                        maxLines = 1,
-                        overflow = TextOverflow.Ellipsis
-                    )
-                }
-            )
+            item {
+                // Signify we have drawn the content of the first screen
+                ReportFullyDrawn()
```
Comment (JohnNichol) [inline ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/ui/landing/LandingScreen.kt:76]: Does this work - do we need it? I am confused as to why this is embedded inside one of the items - should it be a launched effect?

File: ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/ui/userinput/UserInputComponentsScreen.kt [hunk 3]
```diff
@@ -38,41 +32,78 @@ import com.example.android.wearable.composeadvanced.R
...
     value: Int,
     onClickStepper: () -> Unit,
-    onClickSlider: () -> Unit
+    onClickSlider: () -> Unit,
+    onClickDemoDatePicker: () -> Unit,
+    onClickDemo12hTimePicker: () -> Unit,
+    onClickDemo24hTimePicker: () -> Unit,
 ) {
-    Column(
+    ScalingLazyColumn(
```
Comment (JohnNichol) [inline ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/ui/userinput/UserInputComponentsScreen.kt:40]: I don't think we need this

File: ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/ui/userinput/UserInputComponentsScreen.kt [hunk 4]
```diff
@@ -38,41 +32,78 @@ import com.example.android.wearable.composeadvanced.R
...
-                    maxLines = 1,
-                    overflow = TextOverflow.Ellipsis
-                )
-            }
-        )
+        item {
+            CompactChip(
+                onClick = onClickDemoDatePicker,
+                label = {
+                    Text(
```
Comment (JohnNichol) [inline ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/ui/userinput/UserInputComponentsScreen.kt:74]: I would add the current value as a secondaryLabel on a Chip - ditto for next two

File: ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/ui/landing/LandingScreen.kt (hunk 1 above)
Comment (yschimke) [reply ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/ui/landing/LandingScreen.kt:72]: Woah - fillMaxSize() is a default?

File: ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/ui/landing/LandingScreen.kt (hunk 2 above)
Comment (yschimke) [reply ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/ui/landing/LandingScreen.kt:76]: Yep - it's copied from the performance samples.  

https://github.com/android/performance-samples/blob/357887bc8b523ec06d3225ea9920eb6374ac66da/MacrobenchmarkSample/app/src/main/java/com/example/macrobenchmark/target/FullyDrawnStartupActivity.kt

It's there so it's in the frame when the ScalingLazyColumn content is drawn. Not above the ScalingLazyColumn, where it's triggered before a paint with real content.

File: ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/ui/userinput/UserInputComponentsScreen.kt (hunk 3 above)
Comment (yschimke) [reply ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/ui/userinput/UserInputComponentsScreen.kt]: When I changed to Chip from CompactChip, we do.

**Review by JohnNichol — APPROVED** [review]
Thanks Yuri - LGTM

**Pull request merged**